clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -41,7 +41,7 @@ struct CIROpAsmDialectInterface : public OpAsmDialectInterface {
 
   AliasResult getAlias(Type type, raw_ostream &os) const final {
     if (auto structType = type.dyn_cast<StructType>()) {
-      os << structType.getTypeName();
+      os << "ty_" << structType.getTypeName();
```

Comment:
I don't see any change to parsing code, can you also add that? I'm a bit lost on how this is related to https://github.com/llvm/clangir/pull/22, perhaps you might want one PR so you can add tests all at once?

Reply:
The parsing is part of the general MLIR code, the failure is to parse a type alias https://mlir.llvm.org/docs/LangRef/#type-aliases which expect the alias name, that should start with a letter or underscode https://mlir.llvm.org/docs/LangRef/#identifiers-and-keywords.

The dialect just decide what to call those aliases, so the patch makes sure it is a valid name.

Reply:
Relationship with the other PR: to parse back a struct there were two issues, this one which relates to the name of the type alias generated; and the one in #22, that is about parsing the sequence of types that are part of the struct.